src/pages/reports/Reports.tsx

```diff
@@ -1,16 +1,108 @@
-import Strings from "../../utils/Strings";
-import LayoutCard from "../layouts/LayoutCard";
-const Reports = () => {
-    return (
-        <LayoutCard
-            title={Strings.reports}
-            isLoading={false}
-            content={
-                <div className="flex flex-col">
-                   
-                </div>
-            } />
-    );
+import React, { useState } from 'react';
+import { Button, Collapse, DatePicker, Form, Table, Row, Col } from 'antd';
+import LayoutCard from '../layouts/LayoutCard';
+import SectionElement from '../components/SectionElement';
+import Strings from '../../utils/Strings';
+import { usePostReportsMutation } from '../../services/reportsService';
+import { handleErrorNotification } from '../../utils/Notifications';
+import { Reports } from '../../data/reports/report';
+const { Panel } = Collapse;
+
+interface Report {
+	name: string;
+	lastname: string;
+	primaryContact: string;
+	appointment: string;
 }
 
-export default Reports;
\ No newline at end of file
+const ReportsComponent: React.FC = () => {
+	const [started_at, setStarted_at] = useState<string | null>(null);
+	const [finished_at, setFinished_at] = useState<string | null>(null);
+	const [tableData, setTableData] = useState<Report[]>([]);
+	const [postReports] = usePostReportsMutation();
+
+	const handleConsultClick = async () => {
+		try {
+			const result = await postReports({ started_at, finished_at }).unwrap();
+			if (Array.isArray(result)) {
```

Comment:
if you already cast the type data in the service you don't need yo add this

Reply:
I added it because it was the only way I could solve it, but I checked it

Reply:
In the service I made some changes for this feature.

Reply:
you already have this line of code
 transformResponse: (response: { data: Reports}, meta, arg) => response.data

in this case the const  result is the Reports type, so  then you just need to use result.data instead of add Array.isArray(result)

Reply:
I have already removed this lines